tmk_core/protocol/ps2_io_chibios.c

```diff
@@ -0,0 +1,69 @@
+#include <stdbool.h>
+#include "ps2_io.h"
+
+
+// chibiOS headers
+#include "ch.h"
+#include "hal.h"
+
+
+/* Check port settings for clock and data line */
+#if !(defined(PS2_LINE_CLOCK))
+#   error "PS/2 clock setting is required in config.h"
+#endif
+
+#if !(defined(PS2_LINE_DATA))
+#   error "PS/2 data setting is required in config.h"
+#endif
+
+
+
+/*
+ * Clock
+ */
+void clock_init(void)
+{
+}
+
+void clock_lo(void)
+{
+    palSetLineMode(PS2_LINE_CLOCK, PAL_MODE_OUTPUT_OPENDRAIN);
```

Comment:
Does EXT require use of `PAL_MODE_ALTERNATE` on non GPIOv1 boards (for example f303 which is GPIOv2)?

Reply:
good question - dunno :-)
here on GPIOv1/stm32f1 the pins are in no special/alternate mode - probably applies to other stms as well...

Reply:
My understanding is very few boards run GPIOv1, and the majority of QMK supported boards are GPIOv2. I would personally like to see this answered before merging, given the current state of the repo.

Reply:
ok, from what i can tell - grepping through the sources and docs - the EXT driver only uses the pins as they are configured for example with PAL_MODE_OUTPUT_? - no special/alternate mode anywhere. and the EXT driver is available on other STM32F? as well. so i see no problems there

but do note, that the EXT driver is considered legacy - newer chibios versions use per line/pad events that can have a callback function assigned (palEnableLineEvent and palSetLineCallback) - see the comments in my code
yes this adds a minor hurdle when QMK decides to move to a newer chibios version - but i did my best to document that :-)

Reply:
Could I contribute to this project?

Reply:
ChibiOS has been updated since, so this PR may need some TLC before it's ready to merge.